### PR TITLE
[GEARPUMP-367] Don't use windowing unnecessarily

### PR DESCRIPTION
**Pull request by manuzhang**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the commit message is formatted like:
   `[GEARPUMP-<Jira issue #>] Meaningful description of pull request` 
 - [x] Make sure tests pass via `sbt clean test`.
 - [x] Make sure old documentation affected by the pull request has been updated and new documentation added for new functionality.

### CODE REVIEW

File: streaming/src/main/scala/org/apache/gearpump/streaming/dsl/window/impl/WindowRunner.scala [hunk 1]
```diff
@@ -73,6 +73,25 @@ case class AndThen[IN, MIDDLE, OUT](left: WindowRunner[IN, MIDDLE],
   }
 }
 
+class DirectWindowRunner[IN, OUT](fnRunner: FunctionRunner[IN, OUT])
```
Comment (manuzhang) [inline streaming/src/main/scala/org/apache/gearpump/streaming/dsl/window/impl/WindowRunner.scala]: @huafengw I'm a bit stuck for the name here. This is actually not a window runner. Any suggestions ?

Comment (huafengw) [reply streaming/src/main/scala/org/apache/gearpump/streaming/dsl/window/impl/WindowRunner.scala]: Maybe change `WindowRunner` to a more generic name? Then `DirectWindowRunner` can be a non-window runner.

**Comment by manuzhang** [comment]
1. `WindowRunner` => `TimedValueProcessor` which processes `TimestampedValue` and emit `TriggeredOutputs` on trigger
2. Add a subclass `DirectProcessor` which processes `TimestampedValue` with non-folding function (or corresponding `FunctionRunner`
3. Change previous `DefaultWindowRunner` and `WindowRunner`

**Comment by manuzhang** [comment]
R: @huafengw 

The failed test `AppMasterSpec` is unrelated to this PR.

**Comment by huafengw** [comment]
+1

**Pull request closed without merging**